### ISSUE TITLE
Return value type of min/max with key function is lost

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#469]
**Describe the bug**
When using the built-in `min` or `max` function with a key and a known input type, the output type cannot be inferred.

```json
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Inferred type of 'bar' is unknown",
	"source": "pyright (reportUnknownVariableType)",
	"startLineNumber": 4,
	"startColumn": 1,
	"endLineNumber": 4,
	"endColumn": 4
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "No overloads for 'max(foo, key=lambda v: v)' match parameters\n  Argument types: (list[int], (v: Unknown) -> Unknown)",
	"source": "pyright",
	"startLineNumber": 4,
	"startColumn": 7,
	"endLineNumber": 4,
	"endColumn": 32
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Type of 'v' is unknown",
	"source": "pyright (reportUnknownLambdaType)",
	"startLineNumber": 4,
	"startColumn": 27,
	"endLineNumber": 4,
	"endColumn": 28
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Type of lambda expression is unknown",
	"source": "pyright (reportUnknownLambdaType)",
	"startLineNumber": 4,
	"startColumn": 30,
	"endLineNumber": 4,
	"endColumn": 31
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 4,
	"message": "Type of 'bar' is 'Unknown'",
	"source": "pyright",
	"startLineNumber": 5,
	"startColumn": 13,
	"endLineNumber": 5,
	"endColumn": 16
}
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the return value type would be inferred from the input list.

**Screenshots or Code**
```py
# pyright: strict

foo = [1, 2, 3]
bar = max(foo, key=lambda v: v)
reveal_type(bar)
```

**VS Code extension or command-line**
extension, 1.1.17

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is fixed in 1.1.18, which I just published.
